packages/smart-contract/test/smartwalletmanager.ts

```diff
@@ -5,103 +5,105 @@ import { deployContract, solidity } from 'ethereum-waffle';
 import SmartWalletManagerArtifact from '../artifacts/SmartWalletManager.json';
 import { SmartWalletManager } from '../typechain/SmartWalletManager';
 import UserSmartWalletArtifact from '../artifacts/UserSmartWallet.json';
-
-const ZERO = '0x0000000000000000000000000000000000000000';
-const UID = 'u-123';
+import { TransactionResponse } from 'ethers/providers';
+// import { UserSmartWallet } from '../typechain/UserSmartWallet'
 
 chai.use(solidity);
 const { expect } = chai;
 
-describe('SmartWalletManager', () => {
+describe('Testing SmartWalletManager', () => {
   let signers: Signer[];
   let smartWalletManager: SmartWalletManager;
+  // let userSmartWallet: UserSmartWallet;
 
-  beforeEach(async () => {
+  // May need to replace that with `beforeEach` later on:
+  before(async () => {
     signers = await ethers.signers();
     smartWalletManager = (await deployContract(
-      <Wallet>signers[0],
+      <Wallet>signers[0], // `wallet` : If you need to test the contract with a mocked provider
       SmartWalletManagerArtifact,
     )) as SmartWalletManager;
-
-    expect(smartWalletManager.address).to.properAddress;
-    expect(await smartWalletManager.owner()).to.eq(await signers[0].getAddress());
-  });
-
-  it('create a smart-wallet', async () => {
-    const address = await createWallet(smartWalletManager);
-    expect(address).to.not.eq(ZERO);
-    // await expect(contract.method()).to.be.reverted; To-do
   });
 
-  it('fund a smart-wallet', async () => {
-    const address = await createWallet(smartWalletManager);
-    const amount = 1;
-    const tx = await signers[0].sendTransaction({
-      to: address,
-      value: ethers.utils.parseEther(amount.toString()),
+  describe('Deployment tests', async () => {
+    it('Should deploy with a proper contract address', async () => {
```

Comment:
we can adopt best practices early, I saw these one https://yeoman.io/contributing/testing-guidelines.html WDYT ?